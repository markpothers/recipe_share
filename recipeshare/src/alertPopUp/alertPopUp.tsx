import { Modal, Text, View, TouchableOpacity } from "react-native"
import { styles } from "./alertPopupStyleSheet"
import Icon from "react-native-vector-icons/MaterialCommunityIcons"
import {
	responsiveWidth,
	responsiveHeight,
	responsiveFontSize, //eslint-disable-line no-unused-vars
} from "react-native-responsive-dimensions"
import { centralStyles } from "../centralStyleSheet" //eslint-disable-line no-unused-vars
import React from "react"

type Props = {
	title: string;
	close?: () => void;
	closeText?: string;
	onYes: () => void;
	yesText?: string;
};

export function AlertPopup(props: Props) {
	return (
		<Modal animationType="fade" transparent={true} visible={true}>
			<View
				style={[
					styles.modalFullScreenContainer,
					{
						height: responsiveHeight(100),
						width: responsiveWidth(100),
					},
				]}
				accessibilityRole="alert"
			>
				<View style={styles.contentsContainer}>
					<View style={styles.titleContainer}>
						<Text maxFontSizeMultiplier={1.5} style={styles.title}>
							{props.title}
						</Text>
					</View>
					<View
						style={[
							centralStyles.formSection,
							{
								width: responsiveWidth(80),
								marginTop: responsiveHeight(2),
								marginBottom: responsiveHeight(2),
							},
						]}
					>
						<View
							style={[
								centralStyles.formInputContainer,
								{ justifyContent: "space-around" },
							]}
						>
							{props.close && (
								<TouchableOpacity
									style={centralStyles.greenRectangleButton}
									activeOpacity={0.7}
									onPress={props.close}
									testID={"closeButton"}
									accessibilityRole="button"
								>
									<Icon
										style={centralStyles.yellowButtonIcon}
										size={responsiveHeight(4)}
										name="cancel"
									></Icon>
									<Text
										maxFontSizeMultiplier={2}
										style={centralStyles.yellowButtonText}
									>
										{props.closeText
											? props.closeText
											: "Cancel"}
									</Text>
								</TouchableOpacity>
							)}
							{!props.close && (
								<View
									style={[
										centralStyles.greenRectangleButton,
										{ opacity: 0 },
									]}
								></View>
							)}
							<TouchableOpacity
								style={centralStyles.greenRectangleButton}
								activeOpacity={0.7}
								onPress={props.onYes}
								testID={"yesButton"}
								accessibilityRole="button"
							>
								<Icon
									style={centralStyles.yellowButtonIcon}
									size={responsiveHeight(4)}
									name="check"
								></Icon>
								<Text
									maxFontSizeMultiplier={2}
									style={centralStyles.yellowButtonText}
								>
									{props.yesText ? props.yesText : "Yes"}
								</Text>
							</TouchableOpacity>
						</View>
					</View>
				</View>
			</View>
		</Modal>
	)
}
