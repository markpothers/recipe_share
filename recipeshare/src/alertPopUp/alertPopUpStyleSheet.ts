import { StyleSheet } from "react-native"
import { responsiveWidth, responsiveHeight, responsiveFontSize } from "react-native-responsive-dimensions" //eslint-disable-line no-unused-vars

export const styles = StyleSheet.create({
	modalFullScreenContainer: {
		alignItems: "center",
		justifyContent: "center"
	},
	contentsContainer: {
		borderStyle: "solid",
		borderWidth: 1,
		borderColor: "#104e01",
		backgroundColor: "#fff59b",
		width: responsiveWidth(85),
		marginLeft: "7.5%",
		marginRight: "7.5%",
		borderRadius:responsiveWidth(1.5),
		alignItems: "center"
	},
	titleContainer: {
		justifyContent: "center",
		width: "90%",
		paddingTop: responsiveHeight(1)
	},
	title: {
		color: "#104e01",
		fontSize: responsiveFontSize(2.2),
		fontWeight: "bold"
	}
})
