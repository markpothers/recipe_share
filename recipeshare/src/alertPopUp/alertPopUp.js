import React from 'react'
import { Modal, Text, View, TouchableOpacity } from 'react-native'
import { styles } from './alertPopUpStyleSheet'
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import { responsiveWidth, responsiveHeight, responsiveFontSize } from 'react-native-responsive-dimensions'; //eslint-disable-line no-unused-vars
import { centralStyles } from '../centralStyleSheet' //eslint-disable-line no-unused-vars

export function AlertPopUp(props) {
	return (
		<Modal
			animationType="fade"
			transparent={true}
			visible={true}
		>
			<View style={[styles.modalFullScreenContainer, {
				height: responsiveHeight(100),
				width: responsiveWidth(100),
			}]}>
				<View style={styles.contentsContainer}>
					<View style={styles.titleContainer}>
						<Text maxFontSizeMultiplier={1.5} style={styles.title}>{props.title}</Text>
					</View>
					<View style={[centralStyles.formSection, { width: responsiveWidth(80), marginTop: responsiveHeight(2), marginBottom: responsiveHeight(2) }]}>
						<View style={[centralStyles.formInputContainer, { justifyContent: 'space-around' }]}>
							{props.close && (
								<TouchableOpacity style={centralStyles.greenRectangleButton} activeOpacity={0.7} onPress={props.close}>
									<Icon style={centralStyles.yellowButtonIcon} size={responsiveHeight(4)} name='cancel'></Icon>
									<Text maxFontSizeMultiplier={2} style={centralStyles.yellowButtonText}>{props.closeText ? props.closeText : "Cancel"}</Text>
								</TouchableOpacity>
							)}
							{!props.close && (
								<View style={[centralStyles.greenRectangleButton, {opacity: 0}]} >
								</View>
							)}
							<TouchableOpacity style={centralStyles.greenRectangleButton} activeOpacity={0.7} onPress={props.onYes}>
								<Icon style={centralStyles.yellowButtonIcon} size={responsiveHeight(4)} name='check'></Icon>
								<Text maxFontSizeMultiplier={2} style={centralStyles.yellowButtonText}>{props.yesText ? props.yesText : "Yes"}</Text>
							</TouchableOpacity>
						</View>
					</View>
				</View>
			</View>
		</Modal>
	)
}
