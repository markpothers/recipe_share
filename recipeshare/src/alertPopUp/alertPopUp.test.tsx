import React from "react"
import { render, fireEvent } from "@testing-library/react-native"
import { AlertPopUp } from "./alertPopUp"

describe("AlertPopUp", () => {
	let mockClose
	let mockYes

	beforeEach(() => {
		mockClose = jest.fn()
		mockYes = jest.fn()
	})

	test("renders with 2 buttons and customizable text", () => {
		const { toJSON, getAllByRole } = render(
			<AlertPopUp
				close={mockClose}
				closeText={"mock close text"}
				title={"Here is a mock alert question that's being asked"}
				onYes={mockYes}
				yesText={"mock yes text"}
			/>
		)
		expect(toJSON()).toMatchSnapshot()
		const buttons = getAllByRole("button")
		expect(buttons.length).toEqual(2)
	})

	test("renders with 2 buttons and default text", () => {
		const { toJSON, getAllByRole } = render(
			<AlertPopUp
					close={mockClose}
					// closeText={"mock close text"}
					title={"Here is a mock alert question that's being asked"}
					onYes={mockYes}
				// yesText={"Yes button text"}
				/>
		)
		expect(toJSON()).toMatchSnapshot()
		const buttons = getAllByRole("button")
		expect(buttons.length).toEqual(2)
	})

	test("renders with 1 button when no close button and customizable text", () => {
		const { toJSON, getAllByRole } = render(
			<AlertPopUp
					// close={mockClose}
					// closeText={"mock close text"}
					title={"Here is a mock alert question that's being asked"}
					onYes={mockYes}
					yesText={"Yes button text"}
				/>
		)
		expect(toJSON()).toMatchSnapshot()
		const buttons = getAllByRole("button")
		expect(buttons.length).toEqual(1)
	})

	test("calls the yes function when button pressed", () => {
		const { toJSON, getByText } = render(
			<AlertPopUp
				close={mockClose}
				closeText={"mock close text"}
				title={"Here is a mock alert question that's being asked"}
				onYes={mockYes}
				yesText={"mock yes text"}
			/>
		)
		expect(toJSON()).toMatchSnapshot()
		fireEvent.press(getByText("mock yes text"))
		expect(mockYes).toHaveBeenCalled()
		expect(mockYes).toHaveBeenCalledTimes(1)
	})

	test("calls the close function when button pressed", () => {
		const { toJSON, getByText } = render(
			<AlertPopUp
				close={mockClose}
				closeText={"mock close text"}
				title={"Here is a mock alert question that's being asked"}
				onYes={mockYes}
				yesText={"mock yes text"}
			/>
		)
		expect(toJSON()).toMatchSnapshot()
		fireEvent.press(getByText("mock close text"))
		expect(mockClose).toHaveBeenCalled()
		expect(mockClose).toHaveBeenCalledTimes(1)
	})
})
