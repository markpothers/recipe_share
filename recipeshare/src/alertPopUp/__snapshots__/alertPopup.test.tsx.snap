// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertPopup calls the close function when button pressed 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessibilityRole="alert"
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff59b",
          "borderColor": "#104e01",
          "borderRadius": 11.25,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "7.5%",
          "marginRight": "7.5%",
          "width": 637.5,
        }
      }
    >
      <View
        style={
          Object {
            "justifyContent": "center",
            "paddingTop": 13.34,
            "width": "90%",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={1.5}
          style={
            Object {
              "color": "#104e01",
              "fontSize": 33.65552620959067,
              "fontWeight": "bold",
            }
          }
        >
          Here is a mock alert question that's being asked
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "red",
              "borderWidth": 0,
              "overflow": "hidden",
              "width": "100%",
            },
            Object {
              "marginBottom": 26.68,
              "marginTop": 26.68,
              "width": 600,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "purple",
                "borderRadius": 11.25,
                "borderWidth": 0,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginTop": 6.67,
                "minHeight": 80.03999999999999,
                "overflow": "hidden",
                "width": "100%",
              },
              Object {
                "justifyContent": "space-around",
              },
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="closeButton"
          >
            <Icon
              name="cancel"
              size={53.36}
              style={
                Object {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                Object {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              mock close text
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="yesButton"
          >
            <Icon
              name="check"
              size={53.36}
              style={
                Object {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                Object {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              mock yes text
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`AlertPopup calls the yes function when button pressed 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessibilityRole="alert"
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff59b",
          "borderColor": "#104e01",
          "borderRadius": 11.25,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "7.5%",
          "marginRight": "7.5%",
          "width": 637.5,
        }
      }
    >
      <View
        style={
          Object {
            "justifyContent": "center",
            "paddingTop": 13.34,
            "width": "90%",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={1.5}
          style={
            Object {
              "color": "#104e01",
              "fontSize": 33.65552620959067,
              "fontWeight": "bold",
            }
          }
        >
          Here is a mock alert question that's being asked
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "red",
              "borderWidth": 0,
              "overflow": "hidden",
              "width": "100%",
            },
            Object {
              "marginBottom": 26.68,
              "marginTop": 26.68,
              "width": 600,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "purple",
                "borderRadius": 11.25,
                "borderWidth": 0,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginTop": 6.67,
                "minHeight": 80.03999999999999,
                "overflow": "hidden",
                "width": "100%",
              },
              Object {
                "justifyContent": "space-around",
              },
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="closeButton"
          >
            <Icon
              name="cancel"
              size={53.36}
              style={
                Object {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                Object {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              mock close text
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="yesButton"
          >
            <Icon
              name="check"
              size={53.36}
              style={
                Object {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                Object {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              mock yes text
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`AlertPopup renders with 1 button when no close button and customizable text 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessibilityRole="alert"
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff59b",
          "borderColor": "#104e01",
          "borderRadius": 11.25,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "7.5%",
          "marginRight": "7.5%",
          "width": 637.5,
        }
      }
    >
      <View
        style={
          Object {
            "justifyContent": "center",
            "paddingTop": 13.34,
            "width": "90%",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={1.5}
          style={
            Object {
              "color": "#104e01",
              "fontSize": 33.65552620959067,
              "fontWeight": "bold",
            }
          }
        >
          Here is a mock alert question that's being asked
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "red",
              "borderWidth": 0,
              "overflow": "hidden",
              "width": "100%",
            },
            Object {
              "marginBottom": 26.68,
              "marginTop": 26.68,
              "width": 600,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "purple",
                "borderRadius": 11.25,
                "borderWidth": 0,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginTop": 6.67,
                "minHeight": 80.03999999999999,
                "overflow": "hidden",
                "width": "100%",
              },
              Object {
                "justifyContent": "space-around",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#104e01",
                  "borderColor": "#104e01",
                  "borderRadius": 11.25,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "height": "100%",
                  "justifyContent": "space-evenly",
                  "maxWidth": "50%",
                  "minHeight": 80.03999999999999,
                  "minWidth": 285,
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          />
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="yesButton"
          >
            <Icon
              name="check"
              size={53.36}
              style={
                Object {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                Object {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              Yes button text
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`AlertPopup renders with 2 buttons and customizable text 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessibilityRole="alert"
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff59b",
          "borderColor": "#104e01",
          "borderRadius": 11.25,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "7.5%",
          "marginRight": "7.5%",
          "width": 637.5,
        }
      }
    >
      <View
        style={
          Object {
            "justifyContent": "center",
            "paddingTop": 13.34,
            "width": "90%",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={1.5}
          style={
            Object {
              "color": "#104e01",
              "fontSize": 33.65552620959067,
              "fontWeight": "bold",
            }
          }
        >
          Here is a mock alert question that's being asked
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "red",
              "borderWidth": 0,
              "overflow": "hidden",
              "width": "100%",
            },
            Object {
              "marginBottom": 26.68,
              "marginTop": 26.68,
              "width": 600,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "purple",
                "borderRadius": 11.25,
                "borderWidth": 0,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginTop": 6.67,
                "minHeight": 80.03999999999999,
                "overflow": "hidden",
                "width": "100%",
              },
              Object {
                "justifyContent": "space-around",
              },
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="closeButton"
          >
            <Icon
              name="cancel"
              size={53.36}
              style={
                Object {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                Object {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              mock close text
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="yesButton"
          >
            <Icon
              name="check"
              size={53.36}
              style={
                Object {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                Object {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              mock yes text
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`AlertPopup renders with 2 buttons and default text 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessibilityRole="alert"
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff59b",
          "borderColor": "#104e01",
          "borderRadius": 11.25,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "7.5%",
          "marginRight": "7.5%",
          "width": 637.5,
        }
      }
    >
      <View
        style={
          Object {
            "justifyContent": "center",
            "paddingTop": 13.34,
            "width": "90%",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={1.5}
          style={
            Object {
              "color": "#104e01",
              "fontSize": 33.65552620959067,
              "fontWeight": "bold",
            }
          }
        >
          Here is a mock alert question that's being asked
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "red",
              "borderWidth": 0,
              "overflow": "hidden",
              "width": "100%",
            },
            Object {
              "marginBottom": 26.68,
              "marginTop": 26.68,
              "width": 600,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "purple",
                "borderRadius": 11.25,
                "borderWidth": 0,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginTop": 6.67,
                "minHeight": 80.03999999999999,
                "overflow": "hidden",
                "width": "100%",
              },
              Object {
                "justifyContent": "space-around",
              },
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="closeButton"
          >
            <Icon
              name="cancel"
              size={53.36}
              style={
                Object {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                Object {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              Cancel
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="yesButton"
          >
            <Icon
              name="check"
              size={53.36}
              style={
                Object {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                Object {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              Yes
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
